### PR TITLE
Fix batched disposition hang on link disconnects

### LINKED ISSUE #184: Dispositions will hang indefinitely when unsolicitedly terminating with batching enabled
The following code does not handle the case where the disposition buffer is full but the disposition loop is shut down and will hang indefinitely if the condition is 
```go
func (r *Receiver) messageDisposition(ctx context.Context, msg *Message, state encoding.DeliveryState) error {
	var wait chan error
	if r.link.ReceiverSettleMode != nil && *r.link.ReceiverSettleMode == ModeSecond {
		log.Debug(3, "RX (messageDisposition): add %d to inflight", msg.deliveryID)
		wait = r.inFlight.add(msg.deliveryID)
	}

	if r.batching {
		r.dispositions <- messageDisposition{id: msg.deliveryID, state: state}
	} else {
```
Instead, the function should be selecting on this condition or `case <-r.link.Detached:` in order to return when the disposition buffer will no longer be drained.

### PR DESCRIPTION
**Pull request by mcardy**
Fixes #184

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[CHANGELOG.md]: https://github.com/Azure/go-amqp/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by mcardy** [comment]
@microsoft-github-policy-service agree company="solace"

**Comment by mcardy** [comment]
@richardpark-msft can you take a look? Also would it be possible to declare a new release with this fix after its merged? I rely directly on this module. Thanks!

**Comment by jhendrixMSFT** [comment]
I will get this merged and tagged today.

**Comment by jhendrixMSFT** [comment]
Actually, let me back up on the release part.  We've been queueing what's to hopefully be the last of the breaking changes.  We need to review where we're at in that process before we cut the release.

File: receiver.go [hunk 1]
```diff
@@ -294,7 +294,11 @@ func (r *Receiver) messageDisposition(ctx context.Context, msg *Message, state e
 	}
 
 	if r.batching {
-		r.dispositions <- messageDisposition{id: msg.deliveryID, state: state}
+		select {
+		case r.dispositions <- messageDisposition{id: msg.deliveryID, state: state}:
+		case <-r.link.Detached:
+			return ErrLinkClosed
```
Comment (jhendrixMSFT) [inline receiver.go:300]: This should return `r.link.err` instead.

**Comment by mcardy** [comment]
@jhendrixMSFT any update on a new release tag? I'm just curious about the timeline (imminent/next few weeks/not for a while). Thanks!

**Comment by jhendrixMSFT** [comment]
Sorry for the delay.  It's probably 1-2 weeks out.  The last known part of public surface area that needs review/refinement are the various error types.

**Comment by mcardy** [comment]
@jhendrixMSFT thanks for all the hard work on the API improvements lately! I see lots of recent PRs. Do you have a planned date for the release yet?

**Comment by jhendrixMSFT** [comment]
I was discussing this earlier with @richardpark-msft.  Now that all the API changes are in, we're going to run the updated bits through our stress tests for `azeventhubs` and `azservicebus` modules to ensure there are no regressions.  Once that's complete, and things look good, I'll tag the release.  Date-wise I would expect this to be mid-next week.

**Comment by jhendrixMSFT** [comment]
@mcardy FYI `v0.18.0` has been released.

**Pull request merged**